controller/api_impl/metrics-api.go

```diff
@@ -0,0 +1,175 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package api_impl
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/channel"
+	"github.com/openziti/fabric/controller/network"
+	"github.com/openziti/fabric/controller/xmgmt"
+	"github.com/openziti/foundation/identity/identity"
+	"github.com/openziti/xweb/v2"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var _ xweb.ApiHandlerFactory = &MetricsApiFactory{}
+
+type MetricsApiFactory struct {
+	network *network.Network
+	nodeId  identity.Identity
+	xmgmts  []xmgmt.Xmgmt
+}
+
+func (factory *MetricsApiFactory) Validate(_ *xweb.InstanceConfig) error {
+	return nil
+}
+
+func NewMetricsApiFactory(nodeId identity.Identity, network *network.Network, xmgmts []xmgmt.Xmgmt) *MetricsApiFactory {
+	return &MetricsApiFactory{
+		network: network,
+		nodeId:  nodeId,
+		xmgmts:  xmgmts,
+	}
+}
+
+func (factory *MetricsApiFactory) Binding() string {
+	return MetricApiBinding
+}
+
+func (factory *MetricsApiFactory) New(_ *xweb.ServerConfig, options map[interface{}]interface{}) (xweb.ApiHandler, error) {
+
+	metricsApiHandler, err := NewMetricsApiHandler(factory.network, options)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return metricsApiHandler, nil
+}
+
+func NewMetricsApiHandler(n *network.Network, options map[interface{}]interface{}) (*MetricsApiHandler, error) {
+	metricsApi := &MetricsApiHandler{
+		options:    options,
+		network:    n,
+		inspectMgr: network.NewInspectionsManager(n),
+	}
+
```

Comment:
I don't think we do inline certs anywhere else. Should this just be a path to cert file?

Reply:
certainly could be.  I'll look at making that change.